internal/generator/vector/conf_test.go

```diff
@@ -0,0 +1,234 @@
+type = "kubernetes_logs"
+auto_partial_merge = true
+exclude_paths_glob_patterns = ["/var/log/pods/openshift-logging_collector-*/*/*.log", "/var/log/pods/openshift-logging_elasticsearch-*/*/*.log", "/var/log/pods/openshift-logging_kibana-*/*/*.log"]
+
+[sources.raw_journal_logs]
+type = "journald"
+
+# Logs from host audit
+[sources.host_audit_logs]
+type = "file"
+ignore_older_secs = 600
+include = ["/var/log/audit/audit.log"]
+
+# Logs from kubernetes audit
+[sources.k8s_audit_logs]
+type = "file"
+ignore_older_secs = 600
+include = ["/var/log/kube-apiserver/audit.log"]
+
+# Logs from openshift audit
+[sources.openshift_audit_logs]
+type = "file"
+ignore_older_secs = 600
+include = ["/var/log/oauth-apiserver.audit.log"]
+
+[transforms.container_logs]
+type = "remap"
+inputs = ["raw_container_logs"]
+source = '''
+  level = "unknown"
+  if match(.message,r'(Warning|WARN|W[0-9]+|level=warn|Value:warn|"level":"warn")'){
+	level = "warn"
+  } else if match(.message, r'Info|INFO|I[0-9]+|level=info|Value:info|"level":"info"'){
+	level = "info"
+  } else if match(.message, r'Error|ERROR|E[0-9]+|level=error|Value:error|"level":"error"'){
+	level = "error"
+  } else if match(.message, r'Debug|DEBUG|D[0-9]+|level=debug|Value:debug|"level":"debug"'){
+	level = "debug"
+  }
+  .level = level
+
+  .pipeline_metadata.collector.name = "vector"
+  .pipeline_metadata.collector.version = "0.14.1"
+  ip4, err = get_env_var("NODE_IPV4")
+  .pipeline_metadata.collector.ipaddr4 = ip4
+  received, err = format_timestamp(now(),"%+")
+  .pipeline_metadata.collector.received_at = received
+  .pipeline_metadata.collector.error = err
+ '''
+
+[transforms.journal_logs]
+type = "remap"
+inputs = ["raw_journal_logs"]
+source = '''
+  level = "unknown"
+  if match(.message,r'(Warning|WARN|W[0-9]+|level=warn|Value:warn|"level":"warn")'){
+	level = "warn"
+  } else if match(.message, r'Info|INFO|I[0-9]+|level=info|Value:info|"level":"info"'){
+	level = "info"
+  } else if match(.message, r'Error|ERROR|E[0-9]+|level=error|Value:error|"level":"error"'){
+	level = "error"
+  } else if match(.message, r'Debug|DEBUG|D[0-9]+|level=debug|Value:debug|"level":"debug"'){
+	level = "debug"
+  }
+  .level = level
+
+  .pipeline_metadata.collector.name = "vector"
+  .pipeline_metadata.collector.version = "0.14.1"
+  ip4, err = get_env_var("NODE_IPV4")
+  .pipeline_metadata.collector.ipaddr4 = ip4
+  received, err = format_timestamp(now(),"%+")
+  .pipeline_metadata.collector.received_at = received
+  .pipeline_metadata.collector.error = err
+ '''
+
+
+[transforms.route_container_logs]
+type = "route"
+inputs = ["container_logs"]
+route.app = '!(starts_with!(.kubernetes.pod_namespace,"kube") && starts_with!(.kubernetes.pod_namespace,"openshift") && .kubernetes.pod_namespace == "default")'
+route.infra = 'starts_with!(.kubernetes.pod_namespace,"kube") || starts_with!(.kubernetes.pod_namespace,"openshift") || .kubernetes.pod_namespace == "default"'
+
+
+# Rename log stream to "application"
```

Comment:
to bring the conf closer to names defined in our API. pipelines better receive an input named `application` rather than `<some-transform>.<route-name>`
Perhaps in the near future the vrl here will not be empty as we add more features

Reply:
there is lot of scope for optimizing this generated pipeline

Reply:
@vimalk78 my point was the name I originally chose `route.app` was arbitrary.   We can use `application` and then not need to do this step

Reply:
it wont be `application`, it has to be `<component-name>.<route-name>`
and this name needs to pass to the next stage of conf generation. having standard predefined names decouples the stages